lib/cucumber/support_code/step_definition.js

```diff
@@ -106,19 +106,30 @@ function StepDefinition(pattern, options, code) {
 
       Cucumber.Util.Exception.registerUncaughtExceptionHandler(handleException, stepDomain);
 
+      var validCodeLengths = self.validCodeLengths(parameters);
+      if (validCodeLengths.indexOf(code.length) === -1) {
+        var error = 'definition has ' + code.length + ' arguments' +
```

Comment:
It reads even better! 

+1 for the context-dependent names: *step definition* and simply *hook* then I'd say.